src/univariate/continuous/exponential.jl

```diff
@@ -37,7 +42,7 @@ Exponential() = Exponential{Float64}(1.0)
 @distr_support Exponential 0.0 Inf
 
 ### Conversions
-convert(::Type{Exponential{T}}, θ::S) where {T <: Real, S <: Real} = Exponential(T(θ))
+convert(::Type{Exponential{T}}, θ::S) where {T <: Number, S <: Number} = Exponential(T(θ))
```

Comment:
```suggestion
convert(::Type{Exponential{T}}, θ::S) where {T <: Number, S <: Number} = Exponential(convert(T, θ))
```
To support Unitful quantities correctly, `T(θ)` should be changed to `convert(T, θ)`. Otherwise you could convert a unitless number to a unitful `Exponential` (i.e., `convert(Exponential{typeof(1.0m)}, 1.0)`).